### PR TITLE
feat: Upload Modal, photo album image picker and backend upload

### PR DESCRIPTION
**Pull request by yili288**
## Description
Upload Modal for adding images and documents with text.
Image picker from photo album.
Send binary form of image to server using axios.

https://github.com/yili288/readease/assets/47297703/16248da6-e141-4e76-b81f-808dc7a6d362

### CODE REVIEW

File: src/screens/HomePage.tsx [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
 const HomePage = ({ navigation }): JSX.Element => {
-
+    const [uploadModalVisible, setUploadModalVisible] = useState(false);
     return(
         <SafeAreaView style={styles.safeAreacontainer}>
-            <View style={styles.homePageContainer}>
+            <View
+                style={[
+                    styles.homePageContainer,
+                    uploadModalVisible ? { opacity: 0.5 } : {}
```
Comment (yili288) [inline src/screens/HomePage.tsx:26]: greys out the background when the modal is open

**Review by lucasnchoi — APPROVED** [review]
LGTM. Checked out branch worked on my local. 

image

File: src/utils/uploadImage.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+// send images in the form of bytes to the server
+const uploadImage = async (imageUri) => {
+  try {
+    const formData = new FormData();
+    formData.append('uploaded-image', {
+      uri: imageUri,
+      type: 'image/jpg',
+      name: 'image.jpg'
+    });
+    const response = await axios.post(apiUrl + '/image/upload', formData, {
```
Comment (Peteredwardj) [inline src/utils/uploadImage.js]: if we are going with https://github.com/Peteredwardj/readease-api/pull/6 , we can change this endpoint to just `/upload` 🙏

Comment (Peteredwardj) [reply src/utils/uploadImage.js]: Also, the server will return the Text object. So maybe we can handle that somewhere?

Comment (yili288) [reply src/utils/uploadImage.js]: yup sure

**Comment by lucasnchoi** [comment]
merged my PR for the camera and file selection into this branch

**Pull request merged**